functions/_fzf_search_history.fish

```diff
@@ -19,8 +19,15 @@ function _fzf_search_history --description "Search command history. Replace the
     )
 
     if test $status -eq 0
-        set command_selected (string split --max 1 " │ " $command_with_ts)[2]
-        commandline --replace -- $command_selected
+        set commands_selected # empty list to loop over selected commands, in case multiple
+        for c in (string split \n $commands_with_ts)
+            if string match -r --quiet '(\d\d-\d\d \d\d:\d\d:\d\d)' $c
```

Comment:
Great idea! I think this can all be simplified if you just did `string replace --regex` and replaced all instances of that string. I'd also add a `^` so it only matches dates in the beginning of the line.